hawk/app/models/crm_config.rb

```diff
@@ -48,60 +48,71 @@ def persisted?
   end
 
   class << self
+    def get_parameters_from(crm_config, cmd)
+      #todo: this doesn't work with safe_x. research why.
+      REXML::Document.new(%x[#{cmd} 2>/dev/null]).tap do |xml|
+        return unless xml.root
+
+        xml.elements.each("//parameter") do |param|
+          name = param.attributes["name"]
+          content = param.elements["content"]
+          shortdesc = param.elements["shortdesc[@lang=\"#{I18n.locale.to_s.gsub("-", "_")}\"]|shortdesc[@lang=\"en\"]"].text || ""
+          longdesc  = param.elements["longdesc[@lang=\"#{I18n.locale.to_s.gsub("-", "_")}\"]|longdesc[@lang=\"en\"]"].text || ""
+
+          type = content.attributes["type"]
+          default = content.attributes["default"]
+
+          advanced = param.attributes["advanced"] || shortdesc.match(/advanced use only/i) || longdesc.match(/advanced use only/i)
+
+          crm_config[name] = {
+            type: content.attributes["type"],
+            readonly: false,
+            shortdesc: shortdesc,
+            longdesc: longdesc,
+            advanced: (advanced and (advanced=="1")) ? true : false,
+            default: default
+          }
+
+          if type == "enum"
```

Comment:
I'm confused where the type name "enum" is from :-)  It's called "select" type in OCF standard 1.1:

https://github.com/ClusterLabs/OCF-spec/blob/main/ra/1.1/ra-api.rng#L70

Reply:
I didn't change this part but simply moved this block of code. If it should be changed, I will create another PR.